From f749e4a12b30bbe2e71197c2a00a7f7358a355c4 Mon Sep 17 00:00:00 2001
From: Xiexiaobo <X.Xie@freescale.com>
Date: Thu, 18 Jun 2009 16:39:26 +0800
Subject: [PATCH] ENGR00113467 MX25: Fix kernel dump issue when CAN up again

The MX25 gpio_can_inactive() didn't free the D14 GPIO pin, So when
re-up the CAN port, the kernel will dump warning message.

Signed-off-by: Xie Xiaobo <X.Xie@freescale.com>
---
 arch/arm/mach-mx25/mx25_3stack_gpio.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx25/mx25_3stack_gpio.c b/arch/arm/mach-mx25/mx25_3stack_gpio.c
index 83fbc9d..acb9e29 100644
--- a/arch/arm/mach-mx25/mx25_3stack_gpio.c
+++ b/arch/arm/mach-mx25/mx25_3stack_gpio.c
@@ -1175,6 +1175,7 @@ void gpio_can_inactive(int id)
 
 		/* Disable input by setting PWDN/TLE6250.INH high */
 		mxc_set_gpio_dataout(MX25_PIN_D14, 1);
+		mxc_free_iomux(MX25_PIN_D14, MUX_CONFIG_ALT5);
 		break;
 	default:
 		break;
-- 
1.5.4.4

